### DISCUSSION TITLE
String functions on label (search/replace)?

### DISCUSSION THREAD

**Discussion by xxxserxxx** [polybar/polybar#3168] [Q&A]
I'd like my filesystem module to show only the base name of the mount point; I don't want the entire path. `%mountpoint%` is a full path, and I don't find a way to perform string manipulations on this.

What would be most broadly useful would be support for a search/replace, like bash's "Parameter Expansion" functions, but for this specific case, fd's `{}` variations would be useful for most path manipulations: `{}` for the whole path, `{/}` for the basename, `{//}` for the parent, and so on.

If this functionality exists, it'd be _fantastic_ if it were referenced in the Formatting section. If there's a work-around, I was unable to find it. If it's not possible, it'd be a great addition. If this has been discussed and shot down, can someone point me to the discussion so I can understand the reasoning? It seems like desirable, fairly fundamental functionality, and I'm surprised I wasn't able to locate any discussion about it.

**Comment by patrick96** [comment]
I'm a bit hesitant adding sub-languages into the polybar config for this. Would something like adding a new tokent that contains the filesystem label make sense?

For a long time now I have felt that the polybar config format is rather limiting when you want to do more complex things. There has been some effort to get lua support into polybar, which, in the far future, would allow us to run arbitrary lua code to format a token or a label. See also #2945 for discussions. That effort has pretty much stalled though because I just don't have the time to review the changes.

**Reply by xxxserxxx** [reply]
> I'm a bit hesitant adding sub-languages into the polybar config for this. Would something like adding a new tokent that contains the filesystem label make sense?

Yes, absolutely. I really just want to reduce the path; if there were a token that meant `basename %mountpoint%`, that'd solve my need. I was suggesting a more broadly useful solution, but I have no immediate use for such beyond trimming down the mount point label.

> See also #2945 for discussions. That effort has pretty much stalled though because I just don't have the time to review the changes.

Thanks, and yes, that's an even _more_ generally useful feature. It'd also reduce the number of Polybar-specific bash scripts that need to be forked on every update. But, to confirm: a token that returns `basename %mountpoint%` would address my immediate need.

Being a maintainer of a popular project is difficult; if it's _really_ popular, you spend all of your time merging PRs and none of your time doing fun feature implementations, and now it's A Job™. Probably the worst unintended consequence of the Bazaar software model.
